### DISCUSSION TITLE
Enable `organizeImports` on python file save

### DISCUSSION THREAD

**Discussion by cyberksh** [LazyVim/LazyVim#2570] [Q&A] [answered]
On installing the `python` extras, a new keymap is added to organize imports by entering `<leader>co`. How do I enable this code action to run on save?

https://www.lazyvim.org/extras/lang/python#nvim-lspconfig
```lua
--     nvim-lspconfig
-- ...
-- ...
        {
          "<leader>co",
          function()
            vim.lsp.buf.code_action({
              apply = true,
              context = {
                only = { "source.organizeImports" },
                diagnostics = {},
              },
            })
          end,
          desc = "Organize Imports",
        },
-- ...
-- ...
```

**Comment by dpetka2001** [comment, marked as answer]
Try to add the function to an auto-command that runs on `BufWritePre` event.

**Reply by cyberksh** [reply]
https://github.com/astral-sh/ruff-lsp/issues/95#issuecomment-1510021409 Found a solution. Thanks.
